### PR TITLE
Remove apply boiler plate

### PR DESCRIPTION
**Pull request by sonjaer**
## _Read then delete this section_

_- Make sure to use a concise title for the pull-request._

_- Use #patch, #minor or #major in the pull-request title to bump the corresponding version. Otherwise, the patch version
will be bumped. [More details](https://github.com/marketplace/actions/github-tag-bump)_

# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/WelcomeWorkflow.scala [hunk 1]
```diff
@@ -50,11 +50,11 @@ class WelcomeWorkflow extends SdkWorkflow {
     val name = builder.inputOfString("name", "The name for the welcome message")
 
     // uses the workflow input as the task input of the GreetTask
-    val greeting = builder.apply("greet", GreetTask(name)).getOutput("greeting")
+    val greeting = builder.apply("greet",  new GreetTask().withInput("name", name)).getOutput("greeting")
```
Comment (bergman) [inline flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/WelcomeWorkflow.scala:53]:
```suggestion
    val greeting = builder.apply("greet", new GreetTask().withInput("name", name)).getOutput("greeting")
```

File: flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/WelcomeWorkflow.scala [hunk 2]
```diff
@@ -50,11 +50,11 @@ class WelcomeWorkflow extends SdkWorkflow {
...
 
     // uses the output of the GreetTask as the task input of the AddQuestionTask
     val greetingWithQuestion = builder
-      .apply("add-question", AddQuestionTask(greeting))
+      .apply("add-question",  new AddQuestionTask().withInput("greeting", greeting))
```
Comment (bergman) [inline flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/WelcomeWorkflow.scala:57]:
```suggestion
      .apply("add-question", new AddQuestionTask().withInput("greeting", greeting))
```

File: flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/AddQuestionTask.scala [hunk 3]
```diff
@@ -42,16 +42,3 @@ class AddQuestionTask
...
-
-  /** Binds input data to this task
-    *
-    * @param greeting
-    *   the input greeting message
-    * @return
-    *   a transformed instance of this class with input data
-    */
-  def apply(greeting: SdkBindingData): SdkTransform =
-    new AddQuestionTask().withInput("greeting", greeting)
```
Comment (bergman) [inline flytekit-examples-scala/src/main/scala/org/flyte/examples/flytekitscala/AddQuestionTask.scala:56]: I thought of one more reason to not have the `apply`/`of` methods: `SdkTransform.withInput()` has overloads for all the primitive types meaning users can call this task with either an `SdkBindingData` from the output of another task, or with a hard-coded string without having to do `SdkBindingData.ofString("foo")` first. Such overloads are masked by the apply method.

**Comment by narape** [comment]
This PR became obsolete with the introduction of typed inputs

**Pull request closed without merging**